processor/spanmetricsprocessor/processor_test.go

```diff
@@ -291,6 +298,11 @@ func BenchmarkProcessorConsumeTraces(b *testing.B) {
 
 func newProcessorImp(mexp *mocks.MetricsExporter, tcon *mocks.TracesConsumer, defaultNullValue *string) *processorImp {
 	defaultNotInSpanAttrVal := "defaultNotInSpanAttrVal"
+	// use size 2 for LRU cache for testing purpose
+	metricKeyToDimensions, err := NewCache(metricKeyToDimensionsCacheSize)
+	if err != nil {
+		return nil
```

Comment:
I would `panic` with the `err` here rather than swallow the error and have a nil ref panic later.